cmd/snapd-generator/main.c

```diff
@@ -97,3 +91,106 @@ int main(int argc, char **argv)
 	fprintf(f, "WantedBy=local-fs.target\n");
 	return 0;
 }
+
+static bool file_exists(const char *path) {
+	struct stat buf;
+	return stat(path, &buf) == 0;
+}
+
+static bool executable_exists(const char *name) {
+	char *path = getenv("PATH");
+	if (path == NULL) {
+		return false;
+	}
+	char *path_copy SC_CLEANUP(sc_cleanup_string) = NULL;
+	path_copy = sc_strdup(path);
+
+	char *ptr = NULL;
+	char *token = strtok_r(path_copy, ":", &ptr);
+	char fname[PATH_MAX + 1] = { 0 };
+	while (token) {
+		sc_must_snprintf(fname, sizeof fname, "%s/%s", token, name);
+		if (access(fname, X_OK) == 0) {
+			return true;
+		}
+		token = strtok_r(NULL, ":", &ptr);
+	}
+	return false;
+}
+
+int ensure_fusesquashfs_inside_container(const char *normal_dir)
+{
+	// check if we are running inside a container, systemd
+	// provides this file all the way back to trusty if run in a
+	// container
+	if (!file_exists("/run/systemd/container")) {
+		return 0;
+	}
+
+	char *fstype = NULL;
```

Comment:
It's best not to initialize this. Either the code logic below assigns a value _or_ you get an error from the use of an uninitialized variable.

Reply:
Oh, and the type is wrong. It must be `const char *`

Reply:
Done.